landing_page/views.py

```diff
@@ -34,25 +39,26 @@ def search_movies(request):
     query = request.GET.get("query")
 
     if query:
-        # Get all field names of the Movie model
-        movie_fields = [
-            field.name
-            for field in Movie._meta.get_fields()
-            if isinstance(field, (CharField, TextField))
-        ]
-
-        # Construct a list of Q objects for each field
-        q_objects = [Q(**{f"{field}__icontains": query}) for field in movie_fields]
-
-        # Combine all Q objects using OR operator
-        # movies = Movie.objects.filter(*q_objects).distinct()
-        # json_movies = [{"movie": movie} for movie in movies]
-        json_movies = [
-            {"name": "spiderman", "year": 2023},
-            {"name": "batman", "year": 1999},
-        ]
-
-        return render(request, "landing_page/movie.html", {"movies": json_movies})
+        # Construct Q objects for name, director, and release_year fields
+        q_name = Q(name__icontains=query)
+        q_director = Q(director__icontains=query)
+        q_release_year = Q(year__icontains=query)
+
+        # Combine Q objects using OR operator
+        query_filter = q_name | q_director | q_release_year
+
+        # Query movies matching any of the search criteria
+        movies = Movie.objects.filter(query_filter).distinct().order_by("name")
```

Comment:
I think instead of ordering by name it could be more user friendly to order by similarity to what the user typed and also by popularity (based on the imdb_votes field although that hasn't been populated yet).

Reply:
Its ordered by similarity now